switched from sequelize to typeorm

doodyparizada opened the pull request.

doodyparizada commented on config/default.json:
```diff
@@ -6,19 +6,13 @@
-			}
-		}
+		"database": "database.sqlite",
+		"synchronize": true,
```
I want `synchronize: false` and to have a separate runnable file that knows how to generate tables.

nitzantomer replied:
This is part of the `default` config, which (at least for now) is the "dev config".
In a different config file you can set it to `false`, but for development it is easier with it as `true`

doodyparizada commented on scripts/src/index.ts:
```diff
@@ -42,6 +47,9 @@ app.use((err, req, res, next) => {
 });
 
+// initializing db and models
+initModels().then(str => logger.debug(str));
```
msg?

doodyparizada commented on scripts/src/models/index.ts:
```diff
@@ -0,0 +1,47 @@
+import { normalizeError, path } from "../utils";
+
+const dbConfig = Object.assign(getConfig().db);
+if (dbConfig.type === "sqlite" && !/^[./]/.test(dbConfig.database)) {
```
what?

nitzantomer replied:
Well, if using sqlite and the db file path isn't relative/absolute then make the path relative to the root directory of the project.
For example, take what's in our `config/default.json` configuration:
```
...
"database": "database.sqlite",
...
```
The file that will be used will be `PROJECT_ROOT/database.sqlite`

doodyparizada commented on scripts/src/models/index.ts:
```diff
@@ -0,0 +1,47 @@
+}
+
+function createOnConnectedString(options: ConnectionOptions): string {
+	let str = `connected to ${ options.type } server`;
```
msg

doodyparizada commented on scripts/src/models/offers.ts:
```diff
@@ -0,0 +1,185 @@
+	title: string;
+	image: string;
+	description: string;
+};
```
👍

doodyparizada commented on scripts/src/models/offers.ts:
```diff
@@ -0,0 +1,185 @@
+	private _meta: string;
+
+	@Column({ name: "type" })
+	private _type: "spend" | "earn";
```
this enum should be defined outside and used by all places that need it

doodyparizada commented on scripts/src/models/offers.ts:
```diff
@@ -0,0 +1,185 @@
+	}
+
+	public get isUsed(): boolean {
+		return this._isUsed;
```
I'm thinking that we may want usedBy

nitzantomer replied:
Maybe, so we'll add it when/if needed

doodyparizada commented on scripts/src/models/users.ts:
```diff
@@ -0,0 +1,139 @@
+	private _createdDate: Date;
+
+	@Column({ name: "activated_date" })
+	private _activatedDate: Date;
```
reminder: yohay suggested to have a permissions object that each token encapsulates and a user has to allow permissions per token.

doodyparizada commented on scripts/src/models/users.ts:
```diff
@@ -0,0 +1,139 @@
+
+	constructor() {
+		super();
+	}
```
Do you have to define a constructor if all it does is call super?

doodyparizada commented:
I rebased from master. The main conflict is that I added default values to the models which we need to reimplement according to b68153d13485a082638804ee33336b1f34bb08ef
